components/builder-api/doc/api.raml

```diff
@@ -267,6 +267,21 @@ securitySchemes:
                 description: Project does not exist with corresponding projectId
             422:
                 description: Invalid or missing projectId in body
+    /group:
+        /{id}:
+            /promote:
+                /{channel}:
+                    post:
+                        description: Promote every successful project in a job group to the specified channel
+                        responses:
+                            200:
+                                description: Successful promotion
+                            400:
+                                description: ID or channel not provided
+                            404:
+                                description: Group not found
+                            500:
+                                description: Internal server error
```

Comment:
Couple of notes:
1. We don't 'Fail' groups (except in very rare circumstances) if a package build fails. Mostly we will skip any packages (and their deps) that fail, but still complete the group.  
2. If any build failures happen, we will then use the cli tool for promotion.  So we should NOT use failed projects as a reason to not do the promote.  But it would be fine to warn that some projects didn't succeed and did not get promoted.
3. If the group itself got marked as Failed, then it's valid to return a 403 I think.